tests/flytekit/unit/core/test_cache.py

```diff
@@ -167,3 +172,12 @@ def t_cached_explicit_version(a: int) -> int:
     serialized_t_cached_explicit_version = get_serializable_task(OrderedDict(), default_serialization_settings, t_cached_explicit_version)
     assert serialized_t_cached_explicit_version.template.metadata.discoverable == True
     assert serialized_t_cached_explicit_version.template.metadata.discovery_version == "a-version"
+
+
+def test_cache_policy_exception(default_serialization_settings):
+    # Set the address of the ExceptionCachePolicy in the error message so that the test is robust to changes in the
+    # address of the ExceptionCachePolicy class
+    with pytest.raises(ValueError, match=f"Failed to generate version for cache policy <tests.flytekit.unit.core.test_cache.ExceptionCachePolicy object at .*. Please consider setting the version in the Cache definition, e.g. Cache\(version='v1.2.3'\)"):
```

Comment:
Looks like the windows test is not happy about this: https://github.com/flyteorg/flytekit/actions/runs/13337252564/job/37255093692?pr=3135

I'm okay with testing for a part of the message:

```suggestion
    with pytest.raises(ValueError, match="Failed to generate version"):
```